examples/plot_glm_retinal_ganglia.py

```diff
@@ -0,0 +1,421 @@
+
+# create possion GLM instance
+glm_poisson_hist = GLM(distr='poisson',
+                       verbose=False, alpha=0.05,
+                       max_iter=1000, learning_rate=0.2,
+                       score_metric='pseudo_R2',
+                       reg_lambda=1e-7, eta=4.0)
+
+# fitting to a design matrix with spikes history
+glm_poisson_hist.fit(Xdsgn, spikes_binned)
+
+# predict spike counts
+spikes_pred_poissonGLM_hist = glm_poisson_hist.predict(Xdsgn)
+
+# plot
+markerline, _, _ = plt.stem(t_sample, spikes_binned[sample_index])
+markerline.set_markerfacecolor('none')
+plt.plot(t_sample, spikes_pred_poissonGLM[sample_index],
+         color='green', linewidth=2, label='poissonGLM')
+plt.plot(t_sample, spikes_pred_poissonGLM_hist[sample_index],
+         color='orange', linewidth=2, label='poissonGLM_hist')
+
+plt.xlim([t_sample.min(), t_sample.max()])
+plt.title('Spike counts prediction with spikes history using Poisson GLM')
+plt.ylabel('Binned Spike Counts')
+plt.legend()
+plt.show()
+print('Training perf using Pyglmnet score {:.2f}'.format(glm_poisson_hist.score(Xdsgn, spikes_binned)))
+
+#############################################################################
+# **Using both spikes history and spikes-coupled for predicting spike counts**
+
+n_t_filt = 25 # same as before, stimulation history
+n_t_hist = 20 # spikes history
+
+Xdsgn_coupled = [Xstim]
+for cell_num in range(n_cells):
+    spike_time_cell = spike_times[cell_num]
+    spikes_binned_cell, _ = np.histogram(spike_time_cell, t_bins)
+    spikes_padded = np.pad(spikes_binned_cell, (n_t_hist, 0))
+    Xspikes = hankel(spikes_padded[:-n_t_hist], stim[-n_t_hist:])
+    Xdsgn_coupled.append(Xspikes)
+# this design matrix should have a width of
+# n_t_filt + (n_cells * n_t_hist) = 25 + (20 * 4) = 105
+Xdsgn_coupled = np.hstack(Xdsgn_coupled)
+
+# create possion GLM instance
+glm_poisson_coupled = GLM(distr='poisson',
+                          verbose=False, alpha=0.05,
+                          max_iter=1000, learning_rate=0.2,
+                          score_metric='pseudo_R2',
+                          reg_lambda=1e-7, eta=4.0)
+
+# fitting to a design matrix with spikes history
+glm_poisson_coupled.fit(Xdsgn_coupled, spikes_binned)
+
+# predict spike counts
+spikes_pred_poissonGLM_coupled = glm_poisson_coupled.predict(Xdsgn_coupled)
+
+# plot
+markerline, _, _ = plt.stem(t_sample, spikes_binned[sample_index])
+markerline.set_markerfacecolor('none')
+plt.plot(t_sample, spikes_pred_poissonGLM[sample_index],
+         color='green', linewidth=2, label='poissonGLM')
+plt.plot(t_sample, spikes_pred_poissonGLM_coupled[sample_index],
+         color='pink', linewidth=2, label='poissonGLM_couple')
+
+plt.xlim([t_sample.min(), t_sample.max()])
+plt.title('Spike counts prediction with spikes couple using Poisson GLM')
+plt.ylabel('Binned Spike Counts')
+plt.legend()
+plt.show()
+print('Training perf using Pyglmnet score {:.2f}'.\
```

Comment:
This example is way too long. We can't merge it unless it's less than 100 lines :) Can you aim to do that?

Reply:
The code part itself is less than 100 lines. I don't think it can be further trimmed while keeping it good quality.